schema/build-config-example.js

```diff
@@ -36,18 +37,17 @@ fs.writeFileSync(
   "utf8"
 );
 
-const configStringYaml = configStringJson
-  .replace(/JSON\(C\)/, "YAML")
-  .replace(/\n {2}/g, "\n")
-  .replace(/\/\/ /g, "# ")
-  .replace(/(\s*)"([^"]+)":/g, "$1$2:")
-  .replace(/: \{/g, ":")
-  .replace(/\n\}/g, "")
-  .replace(/,\n/g, "\n")
-  .replace(/^\{\n/, "")
-  .replace(/\}$/, "");
+const configStringYaml = YAML.stringify(configExample, {
+  "lineWidth": 0,
+  "defaultStringType": "QUOTE_DOUBLE",
+  "defaultKeyType": "PLAIN"
+})
+  .replace(/^(\s*)[^-\s]+-sub-description: "(.+)"$/gm, "$1# $2")
+  .replace(/^[^-\s]+-description: "(.+)"$/gm, "\n# $1");
```

Comment:
Can these two be combined? Seems the only difference is spaces and "-sub"? How does this look before transformation?

Reply:
This is close to being the same regex used a few lines above for the JSON.

the description should add a new line before
sub-description should keep indentation but no new line.

I can try to unify but only focused on being consistent with the JSON file approach

Reply:
Okay, thanks for pointing that out! I'd forgotten what I did here. :| Looking again now, two thoughts:
1. What about using the same RegExp for JSON and YAML? They're operating on the same data, so that should be straightforward? It'd turn into something like `replace(descriptionRe, "json or yaml replacement text"` (and again for sub-description).
2. Can YAML use the same trick for creating the first line as JSON does? I was going to comment on it being a little unusual to do the string concat in the call to writeFileSync and this change would make the two more consistent and remove the need for that.

Reply:
```
const commentify = (input, commentLiterals) => commentLiterals +
  // eslint-disable-next-line max-len
  " Example markdownlint configuration with all properties set to their default value\n" +
  input
    // eslint-disable-next-line max-len
    .replace(/^(\s*)"?[^-\s]+-sub-description"?: "?([^"\n]+)"?,?$/gm, "$1" + commentLiterals + " $2")
    // eslint-disable-next-line max-len
    .replace(/^(\s*)"?[^-\s]+-description"?: "?([^"\n]+)"?,?$/gm, "\n$1" + commentLiterals + " $2");
```

something like this?